lib/include/openturns/OTtestcode.hxx

```diff
@@ -454,6 +429,596 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
+      Point cdfgrfd(parameter.getDimension());
+      for (UnsignedInteger j = 0; j < parameter.getDimension(); ++ j)
+      {
+        Distribution distributionClone(distribution_);
+        Point param1(parameter);
+        param1[j] += epsilon;
+        distributionClone.setParameter(param1);
+        const Scalar cdf1 = distributionClone.computeCDF(x);
+        Point param2(parameter);
+        param2[j] -= epsilon;
+        distributionClone.setParameter(param2);
+        const Scalar cdf2 = distributionClone.computeCDF(x);
+        cdfgrfd[j] = (cdf1 - cdf2) / (2.0 * epsilon);
+      }
+      LOGTRACE(OSS() << "cdfgrad(fd)=" << cdfgrfd.__str__());
+      assert_almost_equal(cdfGr, cdfgrfd, 1e-5, 1e-5, "wrong cdf gradient for " + distribution_.__repr__());
+    }
+  }
+
+  void checkSampling() const
+  {
+    const Point x = distribution_.getRealization();
+    LOGTRACE(OSS() << "oneRealization=" << x);
+    if (x.getDimension() != distribution_.getDimension())
+      throw TestFailed(OSS() << "wrong realization dimension for " << distribution_);
+    
+    if (distribution_.isContinuous())
+    {
+      if (!distribution_.getRange().contains(x))
+        throw TestFailed(OSS() << "realization not in range for " << distribution_);
+    }
+    else if (distribution_.isDiscrete())
+    {
+      const Sample support(distribution_.getSupport());
+      if (support.find(x) >= support.getSize())
+        throw TestFailed(OSS() << "realization not in support for " << distribution_);
+    }
+
+    if (distribution_.getDimension() == 1)
+    {
+      LOGTRACE(OSS() << "checking sampling with Kolmogorov/ChiSquared ...");
+      UnsignedInteger size = 100;
+      for (UnsignedInteger i = 0; i < 2; ++i)
+      {
+        Bool accepted = true;
+        if (distribution_.isContinuous())
+          accepted = FittingTest::Kolmogorov(distribution_.getSample(size), distribution_).getBinaryQualityMeasure();
+        else if (distribution_.isDiscrete())
+          accepted = FittingTest::ChiSquared(distribution_.getSample(size), distribution_).getBinaryQualityMeasure();
+        LOGTRACE(OSS() << "Kolmogorov test for size " << size << " accepted=" << accepted);
+        if (!accepted)
+          throw TestFailed(OSS() << "kolmogorov test failed for " << distribution_);
+        size *= 10;
+      }
+    }
+  }
+
+  void checkEntropy() const
+  {
+    if (distribution_.getDimension() == 1)
+    {
+      LOGTRACE(OSS() << "checking entropy...");
+      const Scalar entropy = distribution_.computeEntropy();
+      LOGTRACE(OSS() << "entropy=" << entropy);
+      const Scalar entropyMC  = -distribution_.computeLogPDF(distribution_.getSample(entropySamplingSize_)).computeMean()[0];
+      LOGTRACE(OSS() << "entropy(MC)=" << entropyMC);
+      assert_almost_equal(entropy, entropyMC, entropyTolerance_, entropyTolerance_, "entropy " + distribution_.__repr__());
+    }
+  }
+
+  void checkMoments() const
```

Comment:
Very efficient check of the `Distribution` class against the methods in `Sample`: +1.